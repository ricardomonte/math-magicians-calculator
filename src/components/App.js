import React, { useState } from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import Calculate from '../logic/calculate';

function App() {
  const [state, setState] = useState({ total: null, next: null, operator: null });

  const handleClick = (simbolOperation) => {
    setState(Calculate(state, simbolOperation));
  };

  const result = () => {
    const { total, next, operator } = state;
    let display = '0';
    if (next) {
      display = total + operator + next;
    } else if ((operator && operator !== '=') && total) {
      display = total + operator;
    } else {
      display = total || '0';
    }
    return display;
  };

  return (
    <>
      <div className="Math">
        <h2>Let&acute;s do some math </h2>
      </div>
      <div className="w-100 h-100 App">
        <Display calculation={result()} />
        <ButtonPanel onClick={handleClick} />
      </div>
    </>
  );
}

export default App;
