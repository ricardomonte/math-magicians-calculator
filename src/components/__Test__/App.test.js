import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import App from '../App';

test('return content from Home matches snapshot', () => {
  const content = render(<App />);
  expect(content.container).toMatchSnapshot();
  expect(screen.getByText(/LetÂ´s do some math/i)).toBeInTheDocument();
});
