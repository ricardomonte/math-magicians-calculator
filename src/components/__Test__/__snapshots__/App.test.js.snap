// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`return content from Home matches snapshot 1`] = `
<div>
  <div
    class="Math"
  >
    <h2>
      Let´s do some math 
    </h2>
  </div>
  <div
    class="w-100 h-100 App"
  >
    <div
      class="Display hpx-100"
    >
      <h1>
        0
      </h1>
    </div>
    <div
      class="hpx-100"
    >
      <button
        class="btn h-100 bg-button-normal w-25"
        type="button"
      >
        AC
      </button>
      <button
        class="btn h-100 bg-button-normal w-25"
        type="button"
      >
        +/-
      </button>
      <button
        class="btn h-100 bg-button-normal w-25"
        type="button"
      >
        %
      </button>
      <button
        class="btn h-100 bg-button-orange w-25"
        type="button"
      >
        ÷
      </button>
    </div>
    <div
      class="hpx-100"
    >
      <button
        class="btn h-100 bg-button-normal w-25"
        type="button"
      >
        7
      </button>
      <button
        class="btn h-100 bg-button-normal w-25"
        type="button"
      >
        8
      </button>
      <button
        class="btn h-100 bg-button-normal w-25"
        type="button"
      >
        9
      </button>
      <button
        class="btn h-100 bg-button-orange w-25"
        type="button"
      >
        X
      </button>
    </div>
    <div
      class="hpx-100"
    >
      <button
        class="btn h-100 bg-button-normal w-25"
        type="button"
      >
        4
      </button>
      <button
        class="btn h-100 bg-button-normal w-25"
        type="button"
      >
        5
      </button>
      <button
        class="btn h-100 bg-button-normal w-25"
        type="button"
      >
        6
      </button>
      <button
        class="btn h-100 bg-button-orange w-25"
        type="button"
      >
        -
      </button>
    </div>
    <div
      class="hpx-100"
    >
      <button
        class="btn h-100 bg-button-normal w-25"
        type="button"
      >
        1
      </button>
      <button
        class="btn h-100 bg-button-normal w-25"
        type="button"
      >
        2
      </button>
      <button
        class="btn h-100 bg-button-normal w-25"
        type="button"
      >
        3
      </button>
      <button
        class="btn h-100 bg-button-orange w-25"
        type="button"
      >
        +
      </button>
    </div>
    <div
      class="hpx-100"
    >
      <button
        class="btn h-100 bg-button-normal w-50"
        type="button"
      >
        0
      </button>
      <button
        class="btn h-100 bg-button-normal w-25"
        type="button"
      >
        .
      </button>
      <button
        class="btn h-100 bg-button-orange w-25"
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
